# Importing the Libraries

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

# Loading the dataset

In [2]:
buyer_data = pd.read_csv('BuyerRatio.csv')

In [3]:
buyer_data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

###    Let 
###    H0 - The ratio of Male-female buyer are same across all region
###    H1 - The ratio of Male-female buyer are not same across all region

In [9]:
# we will use chi2_contigency because we don't know the distribution and we have to test whether the groups is having same distribution or not
stats.chi2_contingency([buyer_data['East'].value_counts(),buyer_data['West'],buyer_data['North'],buyer_data['South']]) 

(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

### chi2 = 1.5959455386610577
### p-value = 0.6603094907091882
### dof = 3
### expected-ndarray = array([[  42.76531299,  442.23468701],
###                                             [ 146.81287862, 1518.18712138],
###                                             [ 131.11756787, 1355.88243213],
###                                             [  72.30424052,  747.69575948]])

# We can see p-value > alpha value (0.05)
# Therefore, H0 is accepted